tests/test_virtualenv.py

```diff
@@ -370,9 +371,20 @@ def test_copyfile_from_symlink(tmp_path):
     # This may produce a symlink or a regular file depending on the platform --
     # which doesn't matter as long as it has the right contents.
     copy_path = tmp_path / "copy"
-    virtualenv.copyfile(str(src_dir / "link2"), str(copy_path))
+
+    # Check with symlink_resolve True then False
+    virtualenv.copyfile(str(src_dir / "link2"), str(copy_path), symlink_resolve=True)
+    with open(str(copy_path), "r") as f:
+        assert f.read() == "contents"
+    if not virtualenv.IS_WIN:
+        assert os.readlink(str(copy_path)) == str(src_dir / "file")
+
+    os.remove(str(copy_path))
```

Comment:
I think this test should be separate from this 👍 and we should leave untouched this test 👍

Reply:
With theses changes, we get 4 tests for copyfile.
To be sure I understand, do you suggest having two test functions, one using `copyfile` with `symlink_resolve` set to `True` and one where it is `False` ?

Reply:
I would like to have these four test case as separate test functions, and please add a changelog that explains the need and addition.